dss/events/chunkedtask/_awsimpl.py

```diff
@@ -0,0 +1,83 @@
+import typing
+
+import itertools
+
+from ...util.aws import ARN, send_sns_msg
+from . import awsconstants
+from .base import Task, Runtime
+from .constants import TIME_OVERHEAD_FACTOR
+
+
+class AWSRuntime(Runtime[dict]):
+    """
+    This is an implementation of `Runtime` specialized for AWS Lambda.  Work scheduling is done by posting a message
+    containing the serialized state to SNS.
+    """
+    def __init__(self, context, client_name: str) -> None:
+        self.context = context
+        self.client_name = client_name
+
+    def get_remaining_time_in_millis(self) -> int:
+        return self.context.get_remaining_time_in_millis()
+
+    def schedule_work(self, state: dict):
+        sns_arn = ARN(self.context.invoked_function_arn, service="sns", resource=awsconstants.get_worker_sns_topic())
+        send_sns_msg(
+            sns_arn,
+            {
+                awsconstants.CLIENT_KEY: self.client_name,
+                awsconstants.STATE_KEY: state,
+            })
+
+
+# The rest of this file is for unit tests.  The reason they are in this file is because they need to be deployed to the
+# lambda.  For production code, it may make sense to disable this, although it's pretty harmless.
```

Comment:
Consider splitting the test code into its own file to make the organization clearer.